src/generic_sort.cpp

```diff
@@ -373,4 +385,52 @@ std::string GenericDatatypeSort::to_string() const
   return this->compute_string();
 }
 
+DatatypeComponentSort::DatatypeComponentSort(SortKind sk,
+                                             std::string name,
+                                             Sort dt)
+    : GenericSort(sk), name(name), dt_sort(dt)
+{
+  if (sk != CONSTRUCTOR && sk != SELECTOR && sk != TESTER)
+  {
+    throw "Wrong sortkind input";
```

Comment:
It seems like we usually throw `IncorrectUsageException` for those cases:

```suggestion
    throw IncorrectUsageException("Wrong sortkind input");
```